### PR TITLE
 Added Time format Converter

### LINKED ISSUE #7: Addition for time format converter from 12 hour to 24 hour format
Hey, Can I collaborate in your repository?

### PR DESCRIPTION
**Pull request by AvinashSubhash**
#7

### CODE REVIEW

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
@AvinashSubhash Could you remove the extra spaces and lines?

**Review by AvinashSubhash — COMMENTED** [review]
Done

File: Python/time_format_converter.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+        return(str(hour)+s[2:8])
+    elif s[8] == "A":
+        if hour == 12:
+            hour=00
+        return(str(0)+str(hour)+s[2:8])
+
+if __name__ =="__main__":
+    print("Enter the time: (HH:MM:SSPM or HH:MM:SSAM") #Here the last two characters of the string denotes AM or PM"
+    s = input()
+    result = time_conversion(s)
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]: You have missed printing the output here.

File: Python/time_format_converter.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+        if hour<12:
+            hour+=12
+        return(str(hour)+s[2:8])
+    elif s[8] == "A":
+        if hour == 12:
+            hour=00
+        return(str(0)+str(hour)+s[2:8])
+
+if __name__ =="__main__":
+    print("Enter the time: (HH:MM:SSPM or HH:MM:SSAM") #Here the last two characters of the string denotes AM or PM"
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]: Close the bracket and also it might look better with a space between AM and PM and the seconds.

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
There are a few issues that need to be fixed before I can merge it.

**Review by AvinashSubhash — COMMENTED** [review]
I have added the print statement and added a space between seconds and AM/PM in input and output format

File: Python/time_format_converter.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    elif s[9] == "A":
+        if hour == 12:
+            hour=00
+        return(str(0)+str(hour)+s[2:8])
+
+if __name__ =="__main__":
+    print("Enter the time: (HH:MM:SS PM or HH:MM:SS AM)") #Here the last two characters of the string denotes AM or PM"
+    s = input()
+    result = time_conversion(s)
+    print("Converted Time: ",result)
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
    print("Converted Time: ", time_conversion(s))
```

File: Python/time_format_converter.py [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+        if hour<12:
+            hour+=12
+        return(str(hour)+s[2:8])
+    elif s[9] == "A":
+        if hour == 12:
+            hour=00
+        return(str(0)+str(hour)+s[2:8])
+
+if __name__ =="__main__":
+    print("Enter the time: (HH:MM:SS PM or HH:MM:SS AM)") #Here the last two characters of the string denotes AM or PM"
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
    print("Enter the time: (HH:MM:SS PM or HH:MM:SS AM)")
```

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
Nice! You fixed all the issues. But if the user enters `12:09:23`, then the program crashes. Instead, it would be better to ask the user to re-enter a valid input.

**Review by AvinashSubhash — COMMENTED** [review]
The crashing problem has been fixed and statements have been added to check the format of the input

File: Python/time_format_converter.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        return(str(0)+str(hour)+s[2:8])
+    
+    
+if __name__ =="__main__":
+    s = ""
+    print("Enter the time: (HH:MM:SS PM or HH:MM:SS AM)")
+    while len(s) < 11:
+        s = input()
+        if len(s) < 11:
+            print("Invalid Input.Enter the value again!")
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
            print("Invalid Input. Enter the value again!")
```

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
Try entering input as `aaaaaaaaaaa`

**Review by AvinashSubhash — COMMENTED** [review]
I have covered all the conditions :-)

File: Python/time_format_converter.py [hunk 6]
```diff
@@ -0,0 +1,34 @@
+def time_conversion(s):
+
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
```

File: Python/time_format_converter.py [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    else:
+        print("Invalid Input.Enter the value again!")
+    return False
+    
+if __name__ =="__main__":
+    s = ""
+    print("Enter the time: (HH:MM:SS PM or HH:MM:SS AM)")
+    while not validate(s):
+        s = str(input())
+
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
```

File: Python/time_format_converter.py [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        if s[10] == 'M' and (s[9] == 'A' or s[9] == 'P'):
+            if s[2] == ':' and s[5] == ':' and s[8] == ' ':
+                if int(s[0:2]) >=0 and int(s[0:2]) <= 12:
+                    if int(s[3:5]) >=0 and int(s[3:5]) < 60:
+                        if int(s[6:8]) >= 0 and int(s[6:8]) < 60:
+                            return True
+    if flag == True:
+        flag = False
+    else:
+        print("Invalid Input.Enter the value again!")
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
        print("Invalid Input. Please enter in correct format!")
```

File: Python/time_format_converter.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+    return False
+    
+if __name__ =="__main__":
+    s = ""
+    print("Enter the time: (HH:MM:SS PM or HH:MM:SS AM)")
+    while not validate(s):
+        s = str(input())
+        
+    print("Converted Time: ", time_conversion(s))
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
    print("Converted Time:", time_conversion(s))
```

File: Python/time_format_converter.py [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+   
+    hour=int(s[0]+s[1])
+    if s[9] == "P":
+        if hour<12:
+            hour+=12
+        return(str(hour)+s[2:8])
+    elif s[9] == "A":
+        if hour == 12:
+            hour=00
+        return(str(0)+str(hour)+s[2:8])
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
        return(str(0)+str(hour)+s[2:8])

```

File: Python/time_format_converter.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+flag = True
```
Comment (aravindvnair99) [inline Python/time_format_converter.py]:
```suggestion
flag = True
   
```

**Review by aravindvnair99 — APPROVED** [review]
Nice work @AvinashSubhash!

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
@AvinashSubhash  Have a look at the issues pointed out by CodeFactor as well [here](https://github.com/DSC-ASEB/Getting-started-with-Hacktoberfest/pull/8/checks?check_run_id=1237845902).

**Comment by aravindvnair99** [comment]
@AvinashSubhash 👀

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
@AvinashSubhash Please recheck. The output is wrong.

**Comment by aravindvnair99** [comment]
@AvinashSubhash Moving this PR to draft. Once you are done with your changes, you can mark as `Ready for review`.

**Comment by aravindvnair99** [comment]
@AvinashSubhash Reminder: If you fix the pending issues stated above, I will merge this pull request. On the other hand, inactivity will lead to the pull request being closed without merging.

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
@AvinashSubhash Nice work! Try `10:00:00 AM` 10 and 11 have extra zero in the beginning. That's the only bug left, I guess.

**Review by aravindvnair99 — CHANGES_REQUESTED** [review]
@AvinashSubhash Try `12:00:00` series. The output is `None` which is not right.

**Review by aravindvnair99 — APPROVED** [review]

**Pull request merged**